### PR TITLE
Style new const

### PR DESCRIPTION
**Pull request by jaudiger**
If this PR is merged, I'll then be able to constify almost all the methods of "Color": see https://github.com/nushell/nu-ansi-term/pull/61

### CODE REVIEW

File: src/style.rs [hunk 1]
```diff
@@ -262,7 +274,7 @@ impl Style {
     /// assert_eq!(false, Style::default().bold().is_plain());
     /// ```
     pub fn is_plain(self) -> bool {
-        self == Style::default()
+        self == Style::new()
```
Comment (jaudiger) [inline src/style.rs:277]: We cannot constify this method since the equal operand is not usable in const context

**Comment by jaudiger** [comment]
I opened another MR to demonstrate the propagation of const to other structs

**Comment by kubouch** [comment]
Is there some particular reason for constifying the methods (also in your other PRs)? I don't have anything against it, but it's always good to have a concrete use case when contributing something.

**Comment by jaudiger** [comment]
No real use case for the constify of these methods, I had to admit. I just saw an opportunity to rewrite some parts of the code (here through the usage of the "new" method in the default trait) to later enable more methods to be constified. I haven't looked to propagate those new const possibilities elsewhere in the code. But if you are welcome to see more MRs on this topic, I'm open to work on it.

**Comment by kubouch** [comment]
So since there does not seem to be a strong use case, we won't land these const PRs for now. In general, we don't want to change this crate except bugfixes and general maintenance. This crate has some 60M downloads, so any breakage would be quite significant. The const refactor should be harmless, but as there is only one test (if I see it right), we don't want to take any chances.

We could reconsider it if people start complaining that they absolutely need to define const ANSI sequences.

**Pull request closed without merging**